### PR TITLE
Add rel=0 url parameter for youtube embed videos

### PR DESCRIPTION
**Pull request by johangant**
Relates to https://github.com/dof-dss/nidirect-drupal/pull/699

### CODE REVIEW

File: origins_media/origins_media.module [hunk 1]
```diff
@@ -336,3 +336,23 @@ function origins_media_media_library_file_upload_process($form, FormStateInterfa
...
+ * Implements hook_preprocess_media_oembed_iframe().
+ */
+function origins_media_preprocess_media_oembed_iframe(&$variables) {
+  // If we're embedding a Youtube embed object, and there isn't already
+  // and rel=0 parameter... add a rel=0 query parameter to the HTML
+  // to show related videos from the same channel as the video, not videos
+  // based on the viewer's recommendations from their own viewing history.
+  $html = &$variables['media'];
+
+  $is_youtube_embed = preg_match('|youtube.com\/embed|', $html);
```
Comment (omahm) [inline origins_media/origins_media.module:350]: Should we also check against https://youtu.be/ ?

**Review by omahm — COMMENTED** [review]
Looks good, not sure if we need the shortened youtube url (from the share dialog)

**Comment by johangant** [comment]
I've just spotted a link to a drupal.org issue for a patch to core which might well do this + more. If it works I'll favour that over this PR

File: origins_media/origins_media.module (hunk 1 above)
Comment (johangant) [reply origins_media/origins_media.module]: Agreed. Fortunately, core now has a patch which lets us check the provider name (not URL pattern) and has a test which we can borrow the preprocess code from to get the desired result.

**Review by omahm — APPROVED** [review]

**Pull request merged**